File: t/CS/WiaDemo/UI/CapabilitiesListValuesForm.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Leadtools.Wia;
using Leadtools.Demos;

namespace WiaDemo
{
   public partial class CapabilitiesListValuesForm : Form
   {
      public int _selItemIndex;

      public CapabilitiesListValuesForm()
      {
         InitializeComponent();
      }

      private void CapabilitiesListValuesForm_Load(object sender, EventArgs e)
      {
         Int32 value;
         string valueString = string.Empty;
         WiaCapability capability = (WiaCapability)MainForm._capabilitiesList[_selItemIndex];

         _tbPropertyName.Text = capability.PropertyName;

         if ((capability.PropertyAttributes & WiaPropertyAttributesFlags.List) == WiaPropertyAttributesFlags.List)
         {
            // Set the dialog caption.
            this.Text = DemosGlobalization.GetResxString(GetType(), "Resx_WIAListPropertyValues");

            // Change the list control label text.
            _lblValues.Text = DemosGlobalization.GetResxString(GetType(), "Resx_ListValues");

            if (capability.Values.ListValues.ValuesCount <= 0)
               return;

            int index = 0;
            // Add the list values to the list.
            foreach(object i in capability.Values.ListValues.Values)
            {
               if (capability.VariableType == WiaVariableTypes.Bstr)
               {
                  valueString = Convert.ToString(capability.Values.ListValues.Values[index]);
               }
               else if (capability.VariableType == WiaVariableTypes.Clsid)
               {
                  System.Guid guidValue = (System.Guid)capability.Values.ListValues.Values[index];
                  valueString = guidValue.ToString();
               }
               else
               {
                  value = Convert.ToInt32(capability.Values.ListValues.Values[index]);
                  int ret = HelperFunctions.GetWiaListPropertyValueString(capability.PropertyId, (int)value);
                  if (ret == 0)
                  {
                     // Add the received value as is to the values list.
                     valueString = value.ToString();
                  }
                  else
                  {
                     valueString = HelperFunctions.ListPropertyValueString;
                  }
               }
               if (!string.IsNullOrEmpty(valueString))
                  _lbValues.Items.Add(valueString);

               index++;
            }
         }
         else
         {
            // Set the dialog caption.
            this.Text = DemosGlobalization.GetResxString(GetType(), "Resx_FlagPropertyValues");

            // Change the list control label text.
            _lblValues.Text = DemosGlobalization.GetResxString(GetType(), "Resx_FlagValues");

            value = capability.Values.FlagsValues.FlagValues;
            int ret = HelperFunctions.GetWiaFlagPropertyValueString(capability.PropertyId, value);
            if (ret == 0)
            {
               // add the values with their native ID's
               _lbValues.Items.Add(value.ToString());
            }
            else
            {
               foreach (string i in MainForm._flagValuesStrings)
               {
                  _lbValues.Items.Add(i);
               }
            }
         }
      }
   }
}

File: t/PACSFramework/VB/PrintToPACSDemo/Scanning/Wia/CapabilitiesListValuesForm.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports Leadtools.Wia

Namespace PrintToPACSDemo
   Public Partial Class CapabilitiesListValuesForm : Inherits Form
	  Public _selItemIndex As Integer

	  Public Sub New()
		 InitializeComponent()
	  End Sub

	  Private Sub CapabilitiesListValuesForm_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
		 Dim value As Int32
		 Dim valueString As String = String.Empty
		 Dim capability As WiaCapability = CType(FrmMain._capabilitiesList(_selItemIndex), WiaCapability)

		 _tbPropertyName.Text = capability.PropertyName

		 If (capability.PropertyAttributes And WiaPropertyAttributesFlags.List) = WiaPropertyAttributesFlags.List Then
			' Set the dialog caption.
			Me.Text = "WIA List Property Values"

			' Change the list control label text.
			_lblValues.Text = "List Values"

			If capability.Values.ListValues.ValuesCount <= 0 Then
			   Return
			End If

			Dim index As Integer = 0
			' Add the list values to the list.
			For Each i As Object In capability.Values.ListValues.Values
			   If capability.VariableType = WiaVariableTypes.Bstr Then
				  valueString = Convert.ToString(capability.Values.ListValues.Values(index))
			   Else If capability.VariableType = WiaVariableTypes.Clsid Then
				  Dim guidValue As System.Guid = CType(capability.Values.ListValues.Values(index), System.Guid)
				  valueString = guidValue.ToString()
			   Else
				  value = Convert.ToInt32(capability.Values.ListValues.Values(index))
				  Dim ret As Integer = HelperFunctions.GetWiaListPropertyValueString(capability.PropertyId, CInt(value))
				  If ret = 0 Then
					 ' Add the received value as is to the values list.
					 valueString = value.ToString()
				  Else
					 valueString = HelperFunctions.ListPropertyValueString
				  End If
			   End If
			   If (Not String.IsNullOrEmpty(valueString)) Then
				  _lbValues.Items.Add(valueString)
			   End If

			   index += 1
			Next i
		 Else
			' Set the dialog caption.
			Me.Text = "WIA Flag Property Values"

			' Change the list control label text.
			_lblValues.Text = "Flag Values"

			value = capability.Values.FlagsValues.FlagValues
			Dim ret As Integer = HelperFunctions.GetWiaFlagPropertyValueString(capability.PropertyId, value)
			If ret = 0 Then
			   ' add the values with their native ID's
			   _lbValues.Items.Add(value.ToString())
			Else
			   For Each i As String In FrmMain._flagValuesStrings
				  _lbValues.Items.Add(i)
			   Next i
			End If
		 End If
	  End Sub
   End Class
End Namespace
